highfidelity/hifi
# Leopoly_Phase1_003_Asset-Request

... PR staging; description and test plan pending

Changes to the Asset scripting library to support more robust interface for 
managing files in the asset browser

#### (draft) testing plan
1) Using this PR build, launch the local Sandbox and Interface.exe.
2) Go to the localhost sandbox.
3) Open the `Edit -> Assets` browser
    * (if present) delete the `assetUnitTests/` asset folder/files via UI
    * Close the Assets Browser UI.
4) Run the following Client script from URL:
 [/scripts/developer/tests/unit_tests/assetUnitTests.js](https://raw.githubusercontent.com/humbletim/hifi/Leopoly_Phase1_003_Asset-Request/scripts/developer/tests/unit_tests/assetUnitTests.js)
    * A Web Window titled "assetUnitTests.js" should open and launch automated unit tests.
    * Verify that no errors/exceptions are reported (which would be displayed in red).
    * Verify the tests complete successfully (green message):
![image](https://user-images.githubusercontent.com/94753/35287007-6eec7ef8-002f-11e8-97f7-06c65733a5a2.png)
    * notes:
        - it's OK for the placeholder tests to show in yellow at the end
        - some tests generate `"AssetScriptingInterface::downloadData url must be of form atp:<hash-value>"` messages in the main debug log -- that's OK, but in general no other Asset-related errors should appear.
